src/services/TextWriter.hh

```diff
@@ -0,0 +1,31 @@
+#ifndef RA_SERVICES_TEXTWRITER
+#define RA_SERVICES_TEXTWRITER
+#pragma once
+
+#include <string>
+
+namespace ra {
+namespace services {
+
+class TextWriter
+{
+public:
+    virtual ~TextWriter() {}
+
+    /// <summary>
+    /// Writes text to the output.
+    /// </summary>
+    /// <param name="sText">The string to write.</param>
+    virtual void Write(_In_ std::string& sText) = 0;
```

Comment:
In parameters that are references need to be `const `unless you mean `_Inout_ `(param is in a valid state before and after the function but can change).

Reply:
i agree. missed it after copy/pasting TextReader.